### PR TITLE
Ml quality query

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

File: ctapipe/containers.py [hunk 1]
```diff
@@ -700,8 +700,8 @@ class ParticleClassificationContainer(Container):
         nan,
         (
             "prediction of the classifier, defined between "
-            "[0,1], where values close to 0 are more "
-            "gamma-like, and values close to 1 more "
+            "[0,1], where values close to 1 are more "
```
Comment (LukasNickel) [inline ctapipe/containers.py:703]: If I am not missing something, this is now inverted to how the models is getting trained

File: ctapipe/ml/tools/apply_energy_regressor.py [hunk 2]
```diff
@@ -92,7 +92,7 @@ def start(self):
             table = self.loader.read_telescope_events([tel_id])
             table.remove_columns([c for c in table.colnames if c.startswith(prefix)])
 
-            if len(table) == 0:
+            if len(table) == 0 or "hillas_intensity" not in table.colnames:
```
Comment (LukasNickel) [inline ctapipe/ml/tools/apply_energy_regressor.py:95]: I guess this was for testing purposes? Or do you want to assume that column will always be there?

Comment (maxnoe) [reply ctapipe/ml/tools/apply_energy_regressor.py:95]: It is connected to the now fixed issue with impact positions, I used this to check for the empty tables.

File: ctapipe/containers.py (hunk 1 above)
Comment (maxnoe) [reply ctapipe/containers.py:703]: You missed something :D

File: ctapipe/ml/sklearn.py [hunk 3]
```diff
@@ -140,6 +140,7 @@ class Classifier(Model):
     ).tag(config=True)
 
     invalid_class = Integer(-1).tag(config=True)
+    positive_class = Integer(default_value=1).tag(config=True)
```
Comment (maxnoe) [inline ctapipe/ml/sklearn.py:143]: @LukasNickel here

File: ctapipe/ml/apply.py [hunk 4]
```diff
@@ -97,47 +142,111 @@ class RegressionReconstructor(Reconstructor):
...
-    """"""
+    """
+    Predict dl2 particle classification
+    """
+    target = "true_shower_primary_id"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # gammas is parcile_id == 0
+        self.model.positive_class = 0
```
Comment (maxnoe) [inline ctapipe/ml/apply.py:213]: And here @LukasNickel

File: ctapipe/ml/tools/train_particle_classifier.py [hunk 5]
```diff
@@ -119,21 +140,25 @@ def _cross_validate(self, telescope_type, table):
...
+        for (train_indices, test_indices) in tqdm(cv_it, total=n_cv):
             train = table[train_indices]
             test = table[test_indices]
-            self.model.fit(telescope_type, train)
-            prediction, _ = self.model.predict(telescope_type, test)
-            scores.append(metrics.roc_auc_score(test["hadronness"], prediction))
+            self.classifier.model.fit(telescope_type, train)
+            prediction, _ = self.classifier.model.predict_score(telescope_type, test)
+            truth = np.where(
+                test[target] == self.classifier.model.positive_class,
```
Comment (maxnoe) [inline ctapipe/ml/tools/train_particle_classifier.py:157]: And here.

Comment (LukasNickel) [reply ctapipe/ml/tools/train_particle_classifier.py:157]: Okokokok

File: ctapipe/ml/sklearn.py (hunk 3 above)
Comment (LukasNickel) [reply ctapipe/ml/sklearn.py:143]: Wouldn't default value 0 be more reasonable?

Comment (maxnoe) [reply ctapipe/ml/sklearn.py:143]: No, normally you use labels = [0, 1] with one being the positive class.

Comment (LukasNickel) [reply ctapipe/ml/sklearn.py:143]: True.
It is just annoying, that the simtel id for gammas is actually 0 :(

**Comment by LukasNickel** [comment]
pyflakes is not happy
